### PR TITLE
Fix native BlockAlignmentControl

### LINKED ISSUE #35173: Alignment control: Dropdown throws an exception upon opening
## Description

An exception is thrown when opening the block alignment controls on most blocks, the only one that is currently working is the Image block.

**Exception:**
```
ExceptionsManager.js:180 Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Dropdown`.

This error is located at:
    in RCTView (at View.js:34)
    in View (at TouchableHighlight.js:289)
    in TouchableHighlight (at TouchableHighlight.js:352)
    in Unknown (at bottom-sheet/index.native.js:578)
    in RCTScrollContentView (at ScrollView.js:1107)
    in RCTScrollView (at ScrollView.js:1238)
    in ScrollView (at ScrollView.js:1264)
    in ScrollView (at bottom-sheet/index.native.js:554)
    in RCTView (at View.js:34)
    in View (at keyboard-avoiding-view.native.js:118)
    in KeyboardAvoidingView (at bottom-sheet/index.native.js:528)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at createAnimatableComponent.js:599)
    in withAnimatable(View) (created by ReactNativeModal)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at Modal.js:228)
    in RCTView (at View.js:34)
    in View (at Modal.js:249)
    in VirtualizedListContextResetter (at Modal.js:247)
    in RCTModalHostView (at Modal.js:234)
    in Modal (created by ReactNativeModal)
    in ReactNativeModal (at bottom-sheet/index.native.js:499)
    in BottomSheet (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(BottomSheet) (at dropdown-menu/index.native.js:116)
    in Dropdown (at dropdown-menu/index.native.js:68)
    in DropdownMenu (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(DropdownMenu) (at toolbar-dropdown-menu/index.js:17)
    in ToolbarDropdownMenu (at ui.js:158)
    in BlockAlignmentUI (at block-alignment-control/index.js:7)
    in BlockAlignmentControl (at align.js:147)
    in RCTView (at View.js:34)
    in View (at toolbar-group-container.native.js:21)
    in ToolbarGroupContainer (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(ToolbarGroupContainer) (at toolbar-group/index.js:94)
    in ToolbarGroup (at slot.native.js:29)
    in SlotComponent (at slot.js:92)
    in Slot (at slot-fill/index.native.js:16)
    in Slot (at slot-fill/index.native.js:23)
    in BlockControlsBlockSlot (at slot.native.js:24)
    in BlockControlsSlot (at block-toolbar/index.native.js:44)
    in BlockToolbar (at header-toolbar/index.native.js:100)
    in RCTScrollContentView (at ScrollView.js:1107)
    in RCTScrollView (at ScrollView.js:1238)
    in ScrollView (at ScrollView.js:1264)
    in ScrollView (at header-toolbar/index.native.js:89)
    in RCTView (at View.js:34)
    in View (at header-toolbar/index.native.js:83)
    in HeaderToolbar (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(HeaderToolbar) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithViewportMatch(WithPreferredColorScheme(HeaderToolbar)) (at with-dispatch/index.js:96)
    in WithDispatch(WithViewportMatch(WithPreferredColorScheme(HeaderToolbar))) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithDispatch(WithViewportMatch(WithPreferredColorScheme(HeaderToolbar)))) (at header/index.native.js:49)
    in Header (at layout/index.native.js:171)
    in RCTView (at View.js:34)
    in View (at keyboard-avoiding-view/index.ios.js:123)
    in RCTView (at View.js:34)
    in View (at KeyboardAvoidingView.js:209)
    in KeyboardAvoidingView (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at keyboard-avoiding-view/index.ios.js:104)
    in KeyboardAvoidingView (at layout/index.native.js:160)
    in RCTSafeAreaView (at SafeAreaView.js:51)
    in SafeAreaView (at layout/index.native.js:137)
    in RCTView (at View.js:34)
    in View (at tooltip/index.native.js:268)
    in TooltipSlot (at layout/index.native.js:136)
    in Layout (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(Layout) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithPreferredColorScheme(Layout)) (at editor.native.js:173)
    in BlockRefsProvider (at provider/index.native.js:28)
    in BlockEditorProvider (at with-registry-provider.js:24)
    in Unknown (at with-registry/index.js:23)
    in WithRegistryProvider(BlockEditorProvider) (at provider/index.js:108)
    in BlockContextProvider (at provider/index.js:107)
    in EntityProvider (at provider/index.js:106)
    in EntityProvider (at provider/index.js:105)
    in EditorProvider (at with-registry-provider.js:27)
    in Unknown (at with-registry/index.js:23)
    in WithRegistryProvider(EditorProvider) (at provider/index.native.js:337)
    in NativeEditorProvider (at with-dispatch/index.js:96)
    in WithDispatch(NativeEditorProvider) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithDispatch(NativeEditorProvider)) (at editor.native.js:166)
    in SlotFillProvider (at editor.native.js:165)
    in Editor (at with-dispatch/index.js:96)
    in WithDispatch(Editor) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithDispatch(Editor)) (at src/index.native.js:31)
    in App (at renderApplication.js:47)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at renderApplication.js:40)
```

## Step-by-step reproduction instructions

1. Add a Cover block.
2. Tap on the alignment control button located in the toolbar.
Screenshot 2021-09-28 at 10 36 07
3. Observe that an exception is thrown.

## Expected behaviour

The alignment control shouldn't throw any exception.

## Actual behaviour

The alignment control is not working and throws an exception.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/14905380/135053336-f39b6d92-3e2b-4557-8873-748e1b302b81.mp4

## WordPress information
- WordPress version: N/A
- Gutenberg version: N/A
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: iPhone 12 Pro Max
- Operating system: iOS 14.5
- WordPress app version: develop

### PR DESCRIPTION
**Pull request by dcalhoun**
## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/4042

## Description
Fixes #35173. Relates to #34710. A recent web-centric change introduced usage of `MenuGroup` and `MenuItem`, both of which are unsupported in the native editor currently. This updates the conditional within `BlockAlignmentControl` to reinstate the previous UI for the native platform.

### Native Alignment

1. Add a Cover block within the editor. 
2. Press the Alignment icon within the block toolbar. 
3. Change the selected alignment from None to another option.
4. **Expected:** Verify the correct alignment is persisted for the block. 

### Web Alignment
Complete Testing Instructions found in https://github.com/WordPress/gutenberg/pull/34710. 

## Screenshots 
n/a

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-editor/src/components/block-alignment-control/test/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-alignment-control/test/index.native.js]: Since we're getting here the align center button, what do you think about renaming this to `centerAlignmentButton`?

File: packages/block-editor/src/components/block-alignment-control/test/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-alignment-control/test/index.native.js]: Regarding the `when the control is already active` part, do we want to test that `onChange` is called with `undefined` when tapping the currently active alignment value? If this is the case, on line `18` we should get the left alignment button, right?

File: test/native/__mocks__/styleMock.js
Comment (fluiddot) [inline test/native/__mocks__/styleMock.js:140]: Mocking this style is required due to rendering `BlockAlignmentUI` in the `block-alignment-control` test, right?

**Review by fluiddot — COMMENTED** [review]
I verified that the issue is no longer happening with these changes 🎊 , awesome job @dcalhoun 💯 .

The only thing I'd like to double-check before approving the PR is if the `none` alignment value is being properly handled in the Image block. I noticed that the none/default alignment value for that block aligns the content to the left (see attached video):

https://user-images.githubusercontent.com/14905380/135136639-cba0a05a-49a7-4653-86fe-ed15353ce71a.mp4

**Tested on Simulator iPhone 12 Pro Max (iOS 14.5).**

File: test/native/__mocks__/styleMock.js
Comment (dcalhoun) [reply test/native/__mocks__/styleMock.js:140]: Correct. Further down the tree the following code is encountered, which requires a mock of the style attribute referenced directly. https://github.com/WordPress/gutenberg/blob/e5cb098f718d7d1ee1fc2946374c9388e1f0cb2e/packages/components/src/mobile/bottom-sheet/cell.native.js#L412

File: packages/block-editor/src/components/block-alignment-control/test/index.native.js
Comment (dcalhoun) [reply packages/block-editor/src/components/block-alignment-control/test/index.native.js]: Definitely. Thank you for catching this mistake. Addressed in https://github.com/WordPress/gutenberg/pull/35191/commits/bcd16d620af229b66d2f3dd3b22f293552005144.

File: packages/block-editor/src/components/block-alignment-control/test/index.native.js
Comment (dcalhoun) [reply packages/block-editor/src/components/block-alignment-control/test/index.native.js]: Great catch. The odd pairing of relying upon the default left alignment and pressing "None", meant this test passed even though it was not truly the "selected" option. I addressed this in https://github.com/WordPress/gutenberg/pull/35191/commits/bcd16d620af229b66d2f3dd3b22f293552005144 by shifting the test right alignment to be more explicit.

**Comment by dcalhoun** [comment]
> The only thing I'd like to double-check before approving the PR is if the `none` alignment value is being properly handled in the Image block. I noticed that the none/default alignment value for that block aligns the content to the left 

@fluiddot this is a fantastic question. Thanks for bringing this up. 

In regards to explicit code to "handle" a `none` alignment, currently it does not exist. https://github.com/WordPress/gutenberg/blob/bcd16d620af229b66d2f3dd3b22f293552005144/packages/block-library/src/image/edit.native.js#L623-L629

However, I would pose the question: _what should the representation be for the `none` alignment?_ 

From reviewing the web representation, it might imply that representing `none` with a left-aligned image in the editor would be expected, i.e. what we render currently in your example video. The following showcases the `none` and `left` alignment options in both the web editor and web preview. 

<details><summary>Web Editor (top) vs Web Preview (bottom)</summary>

| Alignment: None | Alignment: Left |
| - | - |
| web-editor-none | web-editor-left |
| web-preview-none | web-preview-left |

</details>

Given that, I would say our two options are presenting `none` alignments either centered- or left-aligned. Currently, I think the latter may make the most sense and aligns with the web editor representation. WDYT? 

<details><summary>Native Editor `none` Presentation</summary>

| Center | Left |
| - | - |
| none-option-center | none-option-left |

</details>

**Review by ntsekouras — APPROVED** [review]
Thanks for handling this @dcalhoun ! 💯 

I can see that the main change is `extraProps = isToolbar || Platform.isNative...` so the web version is not affected at all.

Just to learn a bit more, can you please share why MenuGroup and MenuItem are not supported in native? Is there any technical limitation with native?

File: test/native/__mocks__/styleMock.js
Comment (fluiddot) [reply test/native/__mocks__/styleMock.js:140]: Thanks for clarifying this ❤️ !

File: packages/block-editor/src/components/block-alignment-control/test/index.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/block-alignment-control/test/index.native.js]: Thanks for addressing it ❤️ !

I'm thinking that the previous code was covering the new logic of the `None` option. Maybe it's worth it to include that test case (having a selected alignment option and select `None`), wdyt?

**Comment by fluiddot** [comment]
> @fluiddot this is a fantastic question. Thanks for bringing this up.
> 
> In regards to explicit code to "handle" a `none` alignment, currently it does not exist.

Thanks for checking this ❤️ !

> However, I would pose the question: _what should the representation be for the `none` alignment?_
> 
> From reviewing the web representation, it might imply that representing `none` with a left-aligned image in the editor would be expected, i.e. what we render currently in your example video. The following showcases the `none` and `left` alignment options in both the web editor and web preview.
>
> Given that, I would say our two options are presenting `none` alignments either centered- or left-aligned. Currently, I think the latter may make the most sense and aligns with the web editor representation. WDYT?

Great investigation 💯 ! I agree that only center and left alignments fit what I would expect when having `none` alignment value. For the Image block, the current approach with content left-aligned makes sense to me as it matches the same behavior of the web version. It's interesting that this block is the only one that handles the `none` case this way, I noticed that the rest (Embed, Cover, etc.) by default display the content centered.

As a side note, while testing I spotted that the theme might affect the Image block as well, as an example, I used the theme `Spearhead` and the `none` value on the preview is centering the image 🤔 .

Web editor|Web Preview
--|--
Screenshot 2021-09-29 at 10 44 01|Screenshot 2021-09-29 at 10 42 49

File: packages/block-editor/src/components/block-alignment-control/test/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-alignment-control/test/index.native.js]: I removed the comma to match the same format as the following test case's description.
```suggestion
it( 'should call onChange with undefined when the control is already active', () => {
```

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

_Tested `Native Alignment` and `Web Alignment` test cases._

**Comment by mtias** [comment]
Also curious about `MenuGroup` and any timelines for supporting it.

**Comment by dcalhoun** [comment]
> Just to learn a bit more, can you please share why MenuGroup and MenuItem are not supported in native? Is there any technical limitation with native?

> Also curious about `MenuGroup` and any timelines for supporting it.

@ntsekouras @mtias in regards to lack of `MenuGroup` support, there is no technical limitation but likely originates from no specific mobile-centric UI or internal priority requiring it. My general understanding of the focus of internal mobile teams currently largely falls into two categories: increased block support and site/page editing features. So, I would imagine `MenuGroup` has just not been a necessity for those two categories so far. 

Additionally, the mobile UI often leverages more platform-specific UI like our [`Cell` components](https://github.com/WordPress/gutenberg/tree/dad3eefb4fe53cf4fa0dc0729dc5e311a3b84b56/packages/components/src/mobile/bottom-sheet) or native action sheets. There could be opportunity to encapsulate this platform-specific UI as mobile's version of `MenuGroup`/`MenuItem`, but maybe not. It could be the API is so divergent it does not make sense. 

<details><summary>Cell and Action Sheets</summary>

mobile-cell
mobile-cell-2
mobile-action-sheet

</details>

In regards to a timeline for support, I would defer to @hypest, @mchowning, @antonis, and @iamthomasbishop to ask if they have any knowledge of plans to leverage a [`MenuGroup` UI](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/menu-group). 

I interpret the inquiries regarding `MenuGroup` to be non-blocking, so I will move to merge this regression fix to avoid issues in our next mobile release. Thanks. 🙇🏻

**Comment by mtias** [comment]
It's not blocking now, but I want to stay as far away as we can from these platform divergences on shared components, because we are shipping extra code for anyone and just makes maintenance harder.

**Comment by hypest** [comment]
👋 thanks for the ping @dcalhoun!

There are no current plans for supporting `MenuGroup`s, but now it's actually a good time to look into it. David has already pinged you above @iamthomasbishop but yeah, please collaborate with @dcalhoun to have a first assessment of such a project. Thanks!

> I want to stay as far away as we can from these platform divergences on shared components, because we are shipping extra code for anyone and just makes maintenance harder.

I agree with @mtias , it's unfortunate to have to ship and maintain platform specific "forks", especially when they end up being divergent designs. I also understand how they come to be some times: a UI change on the web is merged without being tested on native mobile (at least for crashes) which puts the native mobile team in panic mode to fix the crash in any fast/dirty way they can. It'd be better if we get to collaborate before such merges happen, to at least test out native mobile when new UI is introduced (the "none" alignment option in this case).

**Comment by mtias** [comment]
@hypest I think we need more participation on the UI components side of things as well, since these are primitives that people making updates to blocks should not have to account for because they should have a representation cross-platform.

**Pull request merged**
